### ISSUE TITLE
Provide BackdropCMS releases as tarball

### ISSUE THREAD

**Issue by rfay** [backdrop/backdrop-issues#6675]
## Description of the need

I was working on the DDEV [Backdrop quickstart](https://ddev.readthedocs.io/en/stable/users/quickstart/#backdrop) and noticed that we have to jump through a lot of hoops to get the backdrop files into a named directory. 

I do see
* https://github.com/backdrop/backdrop-issues/issues/4897

But that seems unrelated. 

## Proposed solution

It's so easy to provide a tarball as well as a zipball. 

## Alternatives that have been considered

* The DDEV quickstart can unzip into the parent directory and then rename appropriately, but it might overwrite something in that case.
* We can keep using weird workarounds like we've been doing
  * `unzip ./backdrop.zip && rm backdrop.zip && mv -f ./backdrop/{.,}* . && rm -r backdrop`
  * `unzip ./backdrop.zip && rm -f backdrop.zip && bash -c "mv -f ./backdrop/{.,}* ." && rm -rf backdrop`

zip is just a pretty unwieldy thing, no --strip-components, etc.

## Additional information

Of course, the zipball could not have "backdrop" inside it, but that would break years of tradition, I know.

## Draft of feature description for Press Release (1 paragraph at most)

"Backdrop releases now contain a tarball as well as a zipball. It has the exact same structure."

**Comment by klonos** [comment]
> But that seems unrelated.

Not totally unrelated - one of the arguments in that issue is that we currently only ship .zip release assets (assuming that we will never need to handle tarballs).

Sad that this was raised during a time that missed the dev meeting this week. I'll try to bring it up during next week's dev meeting.

In the meantime, pinging our @backdrop/core-committers for input.

**Comment by rfay** [comment]
If the source tarball were the same as the release, we could just use https://github.com/backdrop/backdrop/tarball/1.28.2 for example (gets backdrop-backdrop-1.28.2-0-g82b5dce.tar.gz). But officialness would be better. 

(Edit: it doesn't really work out, seems the only difference is the official release tag put in the .info files though)

**Comment by klonos** [comment]
> ...seems the only difference is the official release tag put in the .info files though

Yeah, that'd be the packager that does certain things, like removing the various CI/tugboat folders and files and setting the version number. I'll make sure to bring this up during the next dev meeting as I said in my previous comment ...unless one of our core committers chimes in here sooner than that.

Thanks for keep an eye on Backdrop-land @rfay 🙏🏼

**Comment by jenlampton** [comment]
Interesting discussion. In general, we prefer to make things easier on those who are less technical, assuming those who are more technical will be able to find solutions or work-arounds. 

At the time we dropped tarballs, I don't think anyone saw any real benefits to the tarballs over zipballs, and since providing two types caused confusion for most of our users who are less technically-savvy it seemed like an easy decision to make.

Since there do now seem to be reasons to retain tarballs, I don't see any downside to having zipballs avilable on GitHub. We could leave them off backdropcms.org if we still wanted to make things simpler for those using the UIs to grab their downloads.

**Comment by quicksketch** [comment]
Thanks @rfay for reaching out to demonstrate a problem with .zip files. It seems kind of like a silly question, but might there be a better way to unzip the files? (My Googling could not find any working solutions). My previous experience had been that zip files were a lot easier to work with across platforms using a GUI; you can usually double-click them then drag out the files, which wasn't the case with tar.gz files.

A very big difference between `.tar.gz` and `.zip` files is that `.tar.gz` files preserve file permissions, whereas (most) `.zip` files do not. In practice, this means that files receive the default file system permissions with zip archives. That change can trip up developers by having files with different permissions. Though we could address that either by CI-validation to confirm the permissions in Git, or build it into the the packager that we standardize the permissions. There's an issue around this question at https://github.com/backdrop/backdrop-issues/issues/3854

**Comment by rfay** [comment]
I didn't find a good technique beyond what we currently have in the docs, https://ddev.readthedocs.io/en/stable/users/quickstart/#backdrop

`unzip ./backdrop.zip && rm -f backdrop.zip && mv -f ./backdrop/{.,}* . ; rm -rf backdrop
`
UGLY :)

It's pretty easy to set perms to "normal" before tarring them up. And people who need to care about permissions already know what to do.

**Comment by yorkshire-pudding** [comment]
The `bee dl-core` command handles this with the existing zip files. It downloads Backdrop to whatever destination you specify or the current location if you don't. It does this by downloading first to a temp folder and then copying the contents of the folder to the destination and deleting the temp folder.

**Comment by izmeez** [comment]
The `bsdtar` utility also works on zip archives and supports `tar` extract commands including `--strip-components`. It is available for many OS platforms. Example use:

> $ bsdtar --strip-components=1 -zxf /tmp/backdrop.zip --directory "$docroot"

and it can be used in scripts.

**Comment by quicksketch** [comment]
I'm curious if this is still a useful addition to our devops. The DDEV quickstart instructions have been updated to use `bee` directly now:

https://docs.ddev.com/en/stable/users/quickstart/#backdrop

```
mkdir -p my-backdrop-site && cd my-backdrop-site
ddev config --project-type=backdrop
# Add the official Bee CLI add-on
ddev add-on get backdrop-ops/ddev-backdrop-bee
ddev start
# Download Backdrop core
ddev bee download-core
# Create admin user
ddev bee si --username=admin --password=Password123 --db-name=db --db-user=db --db-pass=db --db-host=db --auto
```

Which honestly I think is great that the instructions include installing `bee` because I don't think you'd want to be using DDEV (which is inherently a CLI tool) without also having `bee`.

Our install instructions are still longer than most other projects with DDEV because they depend on `composer`. But the install process is much faster with Backdrop than any Composer-based project which needs to calculate dependencies.

**Comment by izmeez** [comment]
> I'm curious if this is still a useful addition to our devops.

For devops not using DDEV, and possibly even those using DDEV, it may be necessary to have more control when extracting archives. However, having discovered that  `bsdtar` provides the extra control for zip archives as well as tar archives this issue has a solution that also does not conflict with https://github.com/backdrop/backdrop-issues/issues/4897. Other people may have some more thoughts on this and maybe this needs to be added to the documentation?

**Comment by rfay** [comment]
It's definitely true that DDEV doesn't need this for its automated tests any more. We use `ddev bee` for the quickstart tests and docs, and use https://github.com/ddev/test-backdrop with tarball releases for the go-tests. 

So it's an obsolete item from our perspective, only relevant as you see fit for the Backdrop community and CI and such. It's odd to have zipballs and not tarballs, but whatever :) Usually having just zipballs means that a project caters more to Windows than to other environments. But certainly nothing needs to be done any more to cater to DDEV.
